src/acl_hostch.cpp

```diff
@@ -41,7 +41,7 @@ static cl_int l_push_packet(unsigned int physical_device_id, int channel_handle,
   // Pushed data can be smaller than the request write_size due to compiler
   // padding. Runtime needs to check if the trailing bytes are all 0s.
   else if ((pushed_data > 0) && (pushed_data < write_size)) {
-    for (int i = pushed_data; i < write_size; i++) {
+    for (size_t i = pushed_data; i < write_size; i++) {
       unsigned char c = ((char *)host_buffer)[i];
       if (c != 0) {
         // This shouldn't happen. Needs to send out a warning to user rather
```

Comment:
Thanks for cleaning this up Sophie. Could you please modify the code a bit more?

```
        std::cerr << "Error: Data is not fully written into the Hostpipe. "
                     "None-0 bits have been cut off \n";
        assert(0);
```

I wanted to add an assert here since I realize the user would see this error repeatly if they are using a block read, so for now I think we can just force the program to stop if they fall into this path.